pkg/stacks/unpack.go

```diff
@@ -218,13 +238,97 @@ func (sp *StackPackage) IsNamespaced() bool {
 
 // SetApp sets the Stack's App metadata
 func (sp *StackPackage) SetApp(app v1alpha1.AppMetadataSpec) {
-	sp.Stack.Spec.AppMetadataSpec = app
+	app.DeepCopyInto(&sp.Stack.Spec.AppMetadataSpec)
+
 	sp.appSet = true
 }
 
-// SetBehavior sets the Stack's Behavior configuration
-func (sp *StackPackage) SetBehavior(behavior unstructured.Unstructured) {
-	sp.Behavior = behavior
+// SetStackConfig sets the Stack's Behavior using StackConfig
+func (sp *StackPackage) SetStackConfig(config unstructured.Unstructured) {
+	sp.StackConfig = config
+}
+
+func (sp *StackPackage) createBehaviorController(sd v1alpha1.StackDefinitionBehavior) appsv1.DeploymentSpec {
+	// TODO(displague) where does the name come from? behavior? cmd line arg?
+	// sp.Stack.GetName() is not available when this runs
+	controllerDeploymentName := "tc-" + "something"
+	controllerContainerName := controllerDeploymentName
+	controllerImageName := "bymafmaf/templating-controller:0.0.1"
+
+	// TODO(displague) Where do these files reside within the Stack?
+	// are sp.baseDir or sd.Source.Path useful here?
+	// are we always mounting a remote source image?
+	behaviorDirName := "/"
+	behaviorContentsVolumeName := "behaviors"
+	behaviorMountPoint := "/behaviors"
+	spec := appsv1.DeploymentSpec{
+		Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": controllerDeploymentName}},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{"app": controllerDeploymentName},
+			},
+			Spec: corev1.PodSpec{
+				RestartPolicy: corev1.RestartPolicyNever,
+				InitContainers: []corev1.Container{
+					{
+						Name:    "stack-package",
+						Image:   sd.Source.Image,
+						Command: []string{"cp", "-xR", behaviorDirName, behaviorMountPoint},
```

Comment:
In alpine, I get `cp: unrecognized option: x`. Is there any difference if you use `cp -a`?

Reply:
Because I had a root path of `/` I was trying to avoid copying `/proc`, `/sys`, `/dev`, `/tmp`, etc into the mounted volume.  `-x` keeps copy in a single filesystem.

It is a weakness of the current stack packaging system that Stacks must contain a shell at all, and here we see that `cp` within that environment must be a compatible with our expectations.

We can avoid this problem by denying `/` as an image path.

Reply:
Yeah, I think it makes sense just to go with `cp -a`, as far as I see `-a` is compatible with all variants of `cp`. A check for `/` could be helpful before the struct definition.

On another note, I agree that relying on a shell is not the best state to be in but I don't have much solution to overcome this, a `cp` operation needs to be made. Maybe when [CSI drivers ](https://kubernetes-csi.github.io/docs/ephemeral-local-volumes.html)are widely supported, we'd just use the image as a volume for the generic controller and don't need to copy anything.

Reply:
I don't think `cp -a` would work. `-a` tries to set the ownership and mode on the target filesystem.  In Docker build environments I've frequently been snagged by images trying to `chown` or `chmod` volume mounted (or in container) files that the runtime user (non-root) is not permitted to affect or the target volume does not support.

I'll remove the `-x` in anycase, returning the `cp` to its original form and I'll add proper image path usage and validation.